docs/runtimes/functions.md

```diff
@@ -1,27 +1,73 @@
+This workflow pairs especially well with {ref}`ci-cd-automate`.
+		
+## More about functions	
+
+There are various kinds of MLRun functions with different capabilities, however, 
+there are commonalities across all functions. In general, an MLRun function looks like the following:
+
+<img src="../_static/images/mlrun-functions.png" alt="mlrun-architecture" width="700"/><br>
 
 MLRun supports:
-- Real-time functions for: {ref}`serving <serving-graph>`, APIs, and stream processing (based on the high-performance {ref}`Nuclio <nuclio-real-time-functions>` engine). 
-- Batch functions (based on Kubernetes jobs, {ref}`Spark <spark-operator>`, {ref}`Dask <dask-overview>`, {ref}`Horovod <horovod>`, etc.)
+- Real-time runtimes for: {ref}`serving <serving-graph>`, APIs, and stream processing (based on the high-performance {ref}`Nuclio <nuclio-real-time-functions>` engine). 
+- Batch runtimes (based on Kubernetes jobs, {ref}`Spark Operator<spark-operator>`, {ref}`Dask <dask-overview>`, {ref}`Horovod <horovod>`, etc.)
+
+The [different function runtimes](../concepts/functions-overview.html) take care of automatically transforming the code and spec to fully 
+managed and elastic services over Kubernetes, which saves significant operational overhead, 
+addresses scalability, and reduces infrastructure costs. The function parameters and capabilities are explained in more detail in {ref}`create-and-use-functions`.
 
-Function objects are all inclusive (code, spec, API, and metadata definitions), which allows placing them 
-in a shared and versioned function market place. This means that different members of the team can produce or 
-consume functions. Each function is versioned and stored in the MLRun database with a unique hash code, 
+Function objects are all inclusive, containing the code and all the operational aspects: ({ref}`image <images-usage>`, required packages, 
+{ref}`pod resource congifuration (replicas, CPU/GPU/memory limits, volumes, Spot vs. On-demand nodes, pod priority, node affinity) <configuring-job-resources>`, 
+[storage](../runtimes/function-storage.html), environment, metadata definitions, etc.). Each function is versioned and stored in the MLRun database with a unique hash code, 
 and gets a new hash code upon changes.
 
+When you create a function, you specify its name, which appears in the MLRun UI, as well as the Kubernetes pod.
+
+Depending on the job type, the actual python function 
+can be included in the code as a {py:func}`~mlrun.handler`. You can also use the handler to log results and artifacts, see {ref}`decorators-and-auto-logging`.
+
+## Function hub
+
+Since function objects are all-inclusive (code, spec, API, and metadata definitions), they can be stored in, 
+and reused from, a shared and versioned function hub. This means that different members of the team can produce or 
+consume functions. 
```

Comment:
Partially true - it's definitely not easy to share new functions through the hub, and using a custom, private hub, is currently not easily supported. It is easy to consume functions from the hub, maybe we should focus on that...
If we want to emphasize the sharing of functions, maybe it's better to focus on how many users can share the same MLRun project and get access to objects associated with it.

Reply:
...that sentence has been there for ages. Good to get it fixed!